CMakeLists.txt

```diff
@@ -285,6 +298,11 @@ add_custom_target(profile-stats)
 add_custom_target(html-stats)
 add_custom_target(html-profile-stats)
 
+if(PY_SPHINX AND PY_SPHINXCONTRIB.AUTOJINJA AND PY_SPHINX_RTD_THEME AND PY_RECOMMONMARK)
+    message(STATUS "Enabling docs directory as system supports Sphinx builds.")
```

Comment:
Wouldn't it be better to have it as a build option? I think that some people that have these packages installed don't want to build it.

Reply:
That's ok for me. I wanted to make it more convenient and it would automatically build if there is support for it.

The build option can also be triggered by `make docs`

`build_product` would also automatically build the documentation, otherwise we would have to come up with new options to build the documentation.

But I have no strong opinions on making it optional requiring users to run `make docs`.

Reply:
Actually, rephrasing a bit: This `if` is there simply to enable the custom target (`docs`) in CMake (it will include the docs/CMakeLists.txt which contains the custom target to generate docs). If the user's system has these packages installed, the user can run `make docs` after CMake configuration thus generating HTML doc files.